src/components/ProblemSuggestionModal.tsx

```diff
@@ -0,0 +1,331 @@
+  ProblemFeedback,
+} from '../models/problem';
+import className from 'classnames';
+import ButtonGroup from './ButtonGroup';
+import { LANGUAGE_LABELS } from '../context/UserDataContext/properties/userLang';
+import UserDataContext from '../context/UserDataContext/UserDataContext';
+import useProblemSuggestionAction from '../hooks/useProblemSuggestionAction';
+import { useContext } from 'react';
+import MarkdownLayoutContext from '../context/MarkdownLayoutContext';
+import { ModuleInfo } from '../models/module';
+import { SECTION_LABELS } from '../../content/ordering';
+
+export default function ProblemSuggestionModal({
+  isOpen,
+  onClose,
+  tableProblems,
+}: {
+  isOpen: boolean;
+  onClose: Function;
+  tableProblems: Problem[];
+}) {
+  const [name, setName] = React.useState('');
+  const [link, setLink] = React.useState('');
+  const [difficulty, setDifficulty] = React.useState(null);
+  const [tags, setTags] = React.useState('');
+  const [additionalNotes, setAdditionalNotes] = React.useState('');
+
+  const [loading, setLoading] = React.useState(false);
+  const [createdIssueLink, setCreatedIssueLink] = React.useState(null);
+
+  const submitSuggestion = useProblemSuggestionAction();
+  const { markdownLayoutInfo } = useContext(MarkdownLayoutContext);
+
+  React.useEffect(() => {
+    if (isOpen) {
+      setName('');
+      setLink('');
+      setDifficulty(null);
+      setTags('');
+      setAdditionalNotes('');
+      setLoading(false);
+      setCreatedIssueLink(null);
+    }
+  }, [isOpen]);
+
+  const handleSubmit = event => {
+    event.preventDefault();
+
+    setLoading(true);
+
+    // is there a better way to do this? this just identifies the table based on the permalink of the first problem of the table.
+    const problemTableLink =
+      window.location.href.split(/[?#]/)[0] +
+      '#problem-' +
+      tableProblems[0].uniqueID;
+
+    const moduleName = `${
+      SECTION_LABELS[(markdownLayoutInfo as ModuleInfo).section]
+    } - ${markdownLayoutInfo.title}`;
+
+    submitSuggestion({
+      name,
+      link,
+      difficulty,
+      tags,
+      additionalNotes,
+      problemTableLink,
+      moduleName,
+    })
+      .then(response => {
+        setCreatedIssueLink(response.data);
+      })
+      .catch(e => {
+        alert('Error: ' + e.message);
+      })
+      .finally(() => setLoading(false));
+  };
+
+  const form = (
+    <>
+      <div>
+        <label className="block font-medium text-gray-700 dark:text-gray-200">
+          Problem Name
+        </label>
+        <div className="mt-2 relative rounded-md shadow-sm">
+          <input
+            type="text"
+            className="shadow-sm focus:ring-blue-500 focus:border-blue-500 block w-full sm:text-sm border-gray-300 rounded-md dark:bg-gray-900 dark:border-gray-700"
+            placeholder="Ex: USACO December 2012 Silver - Steeplechase"
+            value={name}
+            onChange={e => setName(e.target.value)}
+          />
```

Comment:
I think it's better to make these required like
```suggestion
          <input
            type="text"
            className="shadow-sm focus:ring-blue-500 focus:border-blue-500 block w-full sm:text-sm border-gray-300 rounded-md dark:bg-gray-900 dark:border-gray-700"
            placeholder="Ex: USACO December 2012 Silver - Steeplechase"
            value={name}
            onChange={e => setName(e.target.value)}
            required
          />
```
instead of trying to post the form with missing fields